src/hoverifier.ts

```diff
@@ -811,6 +830,128 @@ export function createHoverifier<C extends object, D, A>({
+                switchMap(({ documentHighlightsOrError, position, adjustPosition, codeView, part, ...rest }) => {
+                    const highlights =
+                        documentHighlightsOrError &&
+                        documentHighlightsOrError !== LOADING &&
+                        !isErrorLike(documentHighlightsOrError)
+                            ? documentHighlightsOrError
+                            : []
+
+                    if (highlights.length === 0 || !position) {
+                        return of({ adjustPosition, codeView, part, ...rest, positions: of<Position[]>([]) })
+                    }
+
+                    const positions: Observable<Position>[] = []
+                    for (const { range } of highlights) {
+                        let pos = { ...position, ...range.start }
+
+                        // The requested position is is 0-indexed; the code here is currently 1-indexed
+                        const { line, character } = pos
+                        pos = { ...pos, line: line + 1, character: character + 1 }
+
+                        positions.push(
+                            adjustPosition
+                                ? from(
+                                      adjustPosition({
+                                          codeView,
+                                          direction: AdjustmentDirection.ActualToCodeView,
+                                          position: {
+                                              ...pos,
+                                              part,
+                                          },
+                                      })
+                                  )
+                                : of(pos)
+                        )
+                    }
+
+                    return of({
+                        adjustPosition,
+                        codeView,
+                        part,
+                        ...rest,
+                        positions: combineLatest(positions),
+                    })
+                }),
+                mergeMap(({ positions, codeView, dom, part }) =>
+                    positions.pipe(
+                        map(highlightedRanges =>
+                            highlightedRanges.map(highlightedRange =>
+                                highlightedRange
+                                    ? getTokenAtPosition(codeView, highlightedRange, dom, part, tokenize)
+                                    : undefined
+                            )
+                        ),
+                        map(elements => ({ elements, codeView, dom, part }))
+                    )
+                )
+            )
+            .subscribe(({ codeView, elements }) => {
+                // Ensure the previously highlighted range is not highlighted and the new highlightedRange (if any)
+                // is highlighted.
+                const currentHighlighteds = codeView.querySelectorAll('.document-highlight')
```

Comment:
I'd pull this out into a constant, and we should probably prefix it with `sourcegraph-` or similar

Reply:
I've only done this to be symmetric with the `selector-highlight` class. Should we do something similar there then?

Reply:
https://github.com/sourcegraph/sourcegraph/blob/36060e86e80fee7b69a30136d5e5bba6c6c5dfb5/browser/src/app.scss#L29

Reply:
Yeah, it's tracked here: https://github.com/sourcegraph/sourcegraph/issues/10595 but not done yet. But it struck my mind while reading it so I thought we could try to not introduce a new instance of it. But maybe it's not necessary, given someone will pick up https://github.com/sourcegraph/sourcegraph/issues/10595 eventually.

Reply:
I'm fine with not adding to the debt as long as there's a decided way to move forward.

Reply:
@felixfbecker do you agree a `sourcegraph-` prefix would be the way we want to push the above mentioned issue forward?

Reply:
Done.

Reply:
We don't need a prefix because in the browser extension we'd just scope the selector. But adding an option for these class names sounds like a good idea to me